Add ACM Private CA section under Network Security

divyansh-gupta opened the pull request:
Adding Application Workload Security page.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

jicowan reviewed (commented):
While this explains how to configure cert-manager to use ACM PCA to issue certificates, it doesn't not explain how to enable TLS between applications running within a cluster. Can you add that information or provide a link to where that information can be found?

jicowan commented on content/security/docs/workloadsecurity.md:
```diff
@@ -0,0 +1,56 @@
+# Application Workload Security
+
+It is important to secure your application workload communications. A main concern is encryption in transit so that your applications and Kubernetes pods can securely communicate with each other. Networking between applications should be secured using TLS.
```
There is a section in networking security where we talk about TLS. Is there a reason you created a separate category for workload security?

dcamzn replied:
We created a separate category following the guidance provided by Micah Hausler during a review of the content.

jicowan replied:
First, thanks for this PR @dcamzn and @divyansh-gupta. I would prefer to move this to the section on [network security/encryption](https://aws.github.io/aws-eks-best-practices/security/docs/network/#encryption-in-transit_1) in transit for now.

dcamzn commented on content/security/docs/workloadsecurity.md:
```diff
@@ -0,0 +1,56 @@
+### Setup Instuctions
+Start by creating a Private CA by following procedures provided in the [ACM Private CA tech docs](https://docs.aws.amazon.com/acm-pca/latest/userguide/create-CA.html). Once you have a Private CA, install cert-manager using [regular installation instructions](https://cert-manager.io/docs/installation/). After installing cert-manager, install the Private CA Kubernetes cert-manager plugin by followi ...
+
+Now that you have a Private CA and an EKS cluster with cert-manager and the plugin installed, it’s time to set permissions and create the issuer. Update IAM permissions of the EKS node role to allow access to ACM Private CA . Replace the `<CA_ARN>` with the value from your Private CA:
```
It makes sense but I am not sure how deep into the instructions we want to go here? Setting up IRSA is covered pretty well by the blog we have linked at the bottom. We could say something like "To see an example of IRSA authentication please see the additional resources blog xyz." Would this work?

dcamzn commented:
> 

How to setup TLS with an AWS load balancer is covered pretty well in the blog as well. If we want to cover in-depth on how to do TLS between applications within a cluster, that feels like it should be it's own blog post. I don't want to blow up this page with those instructions.

divyansh-gupta commented:
Hi @jicowan, thanks for your comments! I went ahead and moved everything under the Network Security section. I also added a line explaining that IRSA can be used, pointing them to an example. I also added 2 blog posts that show how to setup TLS end-to-end. Please let me know what you think. Thanks!

jicowan reviewed (approved):
LGTM! Thanks for the PR.
